impl/src/main/java/com/sun/el/util/ReflectionUtil.java

```diff
@@ -146,15 +144,24 @@ public static PropertyDescriptor getPropertyDescriptor(Object base, Object prope
     /*
      * This method duplicates code in javax.el.ELUtil. When making changes keep the code in sync.
      */
-    public static Object invokeMethod(ELContext context, Method m, Object base, Object[] params) {
+    public static Object invokeMethod(ELContext context, Method method, Object base, Object[] params) {
 
-        Object[] parameters = buildParameters(context, m.getParameterTypes(), m.isVarArgs(), params);
         try {
-            return m.invoke(base, parameters);
+            int paramCount = params == null ? 0 : params.length;
+            if (paramCount == 0) {
+                for (Class ParameterType : method.getParameterTypes()) {
+                    if (ParameterType.getName().equals("javax.faces.event.AjaxBehaviorEvent")) {
+                        return method.invoke(base, (Object[]) params);
+                    }
+                }
+            }
+            Object[] parameters = buildParameters(
+                    context, method.getParameterTypes(), method.isVarArgs(), params);
+            return method.invoke(base, parameters);
         } catch (IllegalAccessException iae) {
             throw new ELException(iae);
         } catch (IllegalArgumentException iae) {
-            throw new ELException(iae);
+            throw new IllegalArgumentException(iae);
```

Comment:
Dtto, explain this change and delete catch block XOR leave original ELException here.